extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerSketchFetcher.java

```diff
@@ -118,22 +120,17 @@ CompletableFuture<Either<Long, ClusterByPartitions>> inMemoryFullSketchMerging(
     final int workerCount = workerTaskIds.size();
     // Guarded by synchronized mergedStatisticsCollector
     final Set<Integer> finishedWorkers = new HashSet<>();
+    final Set<Future<?>> futuresToCancel = ConcurrentHashMap.newKeySet();
```

Comment:
this could be a list - not sure if equals is correctly done for futures

Reply:
also this is not really `futuresToCancel` - this is the exhaustive list of futures

Reply:
Changed